src/jit/lir.cpp

```diff
@@ -525,10 +468,10 @@ GenTree* LIR::Range::FirstNonPhiNode() const
             continue;
         }
 
-        break;
+        return node;
     }
 
-    return node;
+    return nullptr;
```

Comment:
Done. All other suggested changes as that fall out of this method have also been made.